src/download.cpp

```diff
@@ -206,6 +206,13 @@ void DownloadFile(const std::string& url, IDownloadSink *sink, int flags)
 
     if ( !filename_set )
     {
+        char ioptionurl[512];
```

Comment:
If you want to use a fixed buffer, 2048 would be a better size, because it’s the IE/WinInet URL length limit, unlike 512, which is arbitrary. (Getting the length dynamically)[https://github.com/SortByte/Network-Manager/blob/2c80507d3a8f8c73d68df94baf9a0e6a0f698147/Launcher/Network.cpp#L169] would probably be even better, though.